### ISSUE TITLE
Source-bash doesn't honor nested commands in aliases

### ISSUE THREAD

**Issue by jaraco** [xonsh/xonsh#2898]
A common syntax in bash is to pipe the output from one command into another, such as with the following bash alias:

```
alias gpo="git push --set-upstream origin \$(git rev-parse --abbrev-ref HEAD)"
```

Attempting to run that command after running source-bash on it results in this error:

```
$ gpo
error: unknown option `abbrev-ref'
usage: git push [<options>] [<repository> [<refspec>...]]
...
```

## xonfig

```
+------------------+-----------------+
| xonsh            | 0.7.9           |
| Python           | 3.7.1           |
| PLY              | 3.9             |
| have readline    | True            |
| prompt toolkit   | 2.0.4           |
| shell type       | prompt_toolkit2 |
| pygments         | None            |
| on posix         | True            |
| on linux         | False           |
| on darwin        | True            |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

## Expected Behavior

Xonsh should honor this syntax or declare this syntax as unsupported.

## Replication Steps

```
# from any git repo
$ source-bash 'alias gpo="git push --set-upstream origin \$(git rev-parse --abbrev-ref HEAD)"'
$ gpo
```

**Comment by scopatz** [comment]
A simpler example is:

```console
$ source-bash 'alias eee="echo aaa \$(echo b)"'
$ eee
```

**Comment by scopatz** [comment]
Or even just:

```python
aliases['uuu'] = "echo ccc $(echo ddd)"
```

**Comment by scopatz** [comment]
Potential fix in #2956
